pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -215,15 +215,15 @@ func (p *Provider) createLaunchTemplate(ctx context.Context, options *amifamily.
 			},
 			SecurityGroupIds: aws.StringSlice(options.SecurityGroupsIDs),
 			UserData:         aws.String(userData),
-			ImageId:          aws.String(options.AMIID),
+			ImageId:          aws.String(options.Ami.AmiID),
 			MetadataOptions: &ec2.LaunchTemplateInstanceMetadataOptionsRequest{
 				HttpEndpoint:            options.MetadataOptions.HTTPEndpoint,
 				HttpProtocolIpv6:        options.MetadataOptions.HTTPProtocolIPv6,
 				HttpPutResponseHopLimit: options.MetadataOptions.HTTPPutResponseHopLimit,
 				HttpTokens:              options.MetadataOptions.HTTPTokens,
 			},
 			TagSpecifications: []*ec2.LaunchTemplateTagSpecificationRequest{
-				{ResourceType: aws.String(ec2.ResourceTypeNetworkInterface), Tags: utils.MergeTags(options.Tags)},
+				{ResourceType: aws.String(ec2.ResourceTypeInstance), Tags: utils.MergeTags(options.Tags, options.Ami.GetLabelsFromRequirements())},
```

Comment:
I think this is correct and not a type but double-checking with another maintainer who added this code.

Reply:
Please tell me if I'm wrong.
I can add new TagSpecifications if both ResourceTypeNetworkInterface and ResourceTypeInstance are needed.